### PR TITLE
remove duplicated arm and fix version index

### PR DESCRIPTION
**Pull request by xlc**
chatgpt tell me the second arm is unreachable

also explicitly annotate the version index and update the numbers to match xcm version

so the version index for each struct is the xcm version that struct is introduced

Screenshot 2023-03-15 at 9 38 53 PM

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: xcm/src/lib.rs [hunk 1]
```diff
@@ -403,31 +327,39 @@ versioned_type! {
...
 		V2(v2::MultiLocation),
+		#[codec(index = 3)]
 		V3(v3::MultiLocation),
 	}
 }
 
 versioned_type! {
 	/// A single `InteriorMultiLocation` value, together with its version code.
 	pub enum VersionedInteriorMultiLocation {
+		#[codec(index = 2)] // while this is same as v1::Junctions, VersionedInteriorMultiLocation is introduced in v3
```
Comment (xlc) [inline xcm/src/lib.rs:340]: this could be 1. let me know

**Comment by xlc** [comment]
@bkchr we need to adjust the label

File: xcm/src/lib.rs (hunk 1 above)
Comment (KiChjang) [reply xcm/src/lib.rs:340]: Was there a `VersionedJunction` in XCMv2? If so, then it should be 1, otherwise this is ok.

Comment (xlc) [reply xcm/src/lib.rs:340]: Screenshot 2023-03-16 at 12 16 25 AM

all the versioned type in v0.9.37

File: xcm/src/lib.rs [hunk 2]
```diff
@@ -300,101 +303,22 @@ macro_rules! versioned_type {
...
 	pub enum VersionedAssetId {
+		#[codec(index = 3)]
 		V3(v3::AssetId),
 	}
 }
 
 versioned_type! {
 	/// A single version's `Response` value, together with its version code.
 	pub enum VersionedResponse {
+		#[codec(index = 2)]
```
Comment (gavofyork) [inline xcm/src/lib.rs:319]: Is this definitely correct?

Comment (xlc) [reply xcm/src/lib.rs:319]: https://github.com/paritytech/polkadot/blob/645723987cf9662244be8faf4e9b63e8b9a1b3a3/xcm/src/lib.rs#L127-L130

**Review by KiChjang — APPROVED** [review]

**Comment by gavofyork** [comment]
Worth putting in a test?

**Review by gavofyork — APPROVED** [review]

**Comment by xlc** [comment]
I can add some tests tomorrow

**Comment by joepetrowski** [comment]
bot merge

**Comment by abebeos** [comment]
> chatgpt tell me the second arm is unreachable

Is this a joke?

**Comment by SaltyCucumber** [comment]
> > chatgpt tell me the second arm is unreachable
> 
> Is this a joke?

Are you questioning our AI overlord?

**Comment by abebeos** [comment]
> > > chatgpt tell me the second arm is unreachable
> > 
> > Is this a joke?
> 
> Are you questioning our AI overlord?

Nah, just paritytech's project-processes.

**Comment by xlc** [comment]
@bkchr can we backport this to v0.9.38 and v0.9.39

**Comment by bkchr** [comment]
> @bkchr can you backport this to v0.9.38 and v0.9.39

It should already been worked on.

**Comment by crystalin** [comment]
Can we check the labels please? it seems to introduce breaking changes

**Pull request merged**